service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOController.java

```diff
@@ -10,33 +10,49 @@
 import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
 import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
 import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
 import uk.gov.hmcts.reform.fpl.model.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.service.DirectionHelperService;
 import uk.gov.hmcts.reform.fpl.service.DraftCMOService;
 
-import java.util.Map;
+import static java.util.Objects.isNull;
 
 @Api
 @RestController
 @RequestMapping("/callback/draft-cmo")
 public class DraftCMOController {
     private final ObjectMapper mapper;
     private final DraftCMOService draftCMOService;
+    private final DirectionHelperService directionHelperService;
+
 
     @Autowired
-    public DraftCMOController(ObjectMapper mapper, DraftCMOService draftCMOService) {
+    public DraftCMOController(ObjectMapper mapper,
+                              DraftCMOService draftCMOService,
+                              DirectionHelperService directionHelperService) {
         this.mapper = mapper;
         this.draftCMOService = draftCMOService;
+        this.directionHelperService = directionHelperService;
     }
 
     @PostMapping("/about-to-start")
     public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
-        Map<String, Object> caseData = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        if (!isNull(caseData.getCaseManagementOrder())) {
+            directionHelperService.sortDirectionsByAssignee(caseData.getCaseManagementOrder().getDirections())
+                .forEach((key, value) -> caseDetails.getData().put(key.getValue(), value));
+        } else {
+            removeExistingCustomDirections(caseDetails);
```

Comment:
Why you need to remove custom directions when CMO is not there? Why would custom directions be in case data if CMO is not there?

Reply:
Because customDirection are shared between SDO and CMO